pkg/pool-gyro/contracts/GyroECLPPool.sol

```diff
@@ -139,17 +139,25 @@ contract GyroECLPPool is IGyroECLPPool, BalancerPoolToken, PoolInfo, Version {
             require(invariant.x <= GyroECLPMath._MAX_INVARIANT, GyroECLPMath.MaxInvariantExceeded());
         }
 
+        int256 newBalanceInt;
+
         if (tokenInIndex == 0) {
-            return
-                GyroECLPMath
-                    .calcXGivenY(balancesLiveScaled18[1].toInt256(), eclpParams, derivedECLPParams, invariant)
-                    .toUint256();
+            (newBalanceInt, , ) = GyroECLPMath.calcXGivenY(
+                balancesLiveScaled18[1].toInt256(),
+                eclpParams,
+                derivedECLPParams,
+                invariant
+            );
         } else {
-            return
-                GyroECLPMath
-                    .calcYGivenX(balancesLiveScaled18[0].toInt256(), eclpParams, derivedECLPParams, invariant)
-                    .toUint256();
+            (newBalanceInt, , ) = GyroECLPMath.calcYGivenX(
+                balancesLiveScaled18[0].toInt256(),
+                eclpParams,
+                derivedECLPParams,
+                invariant
+            );
         }
+
+        newBalance = newBalanceInt.toUint256();
```

Comment:
I think I'd prefer `return newBalanceInt.toUint256();` given that this function is a bit long.

For the other cases where we now return `a` and `b` I think implicit returns are acceptable.